app/components/Settings/UserConnections/filedrop.jsx

```diff
@@ -0,0 +1,175 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import {SAMPLE_DBS} from '../../../constants/constants';
+
+export default class Filedrop extends Component {
+    static propTypes = {
+        settings: PropTypes.object,
+        connection: PropTypes.object,
+        updateConnection: PropTypes.func,
+        sampleCredentialsStyle : PropTypes.object
+    }
+
+    /**
+     * Filedrop is an input component where users can type an URL or drop a file
```

Comment:
General Comment.  I like the style how you have documented the JSDocs and React class structure.  I have noticed that there seems to be some inconsistency with the React Classes.  Do we have a standard format?  I have noticed some classes have PropTypes at the bottom.  Is there a standard?  Should we document this somewhere?  If so where is a good place?

Reply:
@shannonlal 

>  I have noticed that there seems to be some inconsistency with the React Classes. Do we have a standard format? I have noticed some classes have PropTypes at the bottom. Is there a standard? Should we document this somewhere? If so where is a good place?

We don't have any guidelines. I'd keep them in a file: `CODING_GUIDELINES.md`.

I don't want the guidelines to become a burden (it's easier for us to tell contributors to use given files as models; e.g: we could chose one of the files in the project as a model for stateless components, another file for stateful components, and another for components that use redux). Also, by burden, I mean:
- For example, the burden of prescribing that `PropTypes` go at the top is OK, because it's just a copy'n'paste.
- But, for example, I don't think the burden of prescribing all React components should be class components is worth it (i.e. it's OK, and some would say better, that stateless components are written as [functional components](https://reactjs.org/docs/components-and-props.html#stateless-functions)).